frontend/src/components/Search/index.tsx

```diff
@@ -0,0 +1,262 @@
+        if (!v) {
+          return;
+        }
+        navigate(v.link);
+      }}
+    >
+      <div className="relative w-full">
+        <Icon
+          path={search}
+          className="absolute left-2 top-2 z-10 size-5 text-gray-600"
+        />
+        <ComboboxInput
+          ref={inputRef}
+          displayValue={(result: SearchResult) =>
+            (result || {}).displayTitle || ""
+          }
+          onChange={(event) => onChange(event.target.value)}
+          placeholder={placeholder}
+          className={
+            barSize === "small"
+              ? "relative block h-9 w-96 appearance-none border border-transparent bg-gray-200 px-4 pl-8 text-inherit placeholder:text-gray-500 focus:border-brand-700 focus:outline-none dark:bg-gray-800"
+              : "relative block h-12 w-full appearance-none border border-transparent bg-gray-200 px-4 pl-8 text-inherit placeholder:text-gray-500 focus:border-brand-700 focus:outline-none dark:bg-gray-800"
+          }
+        />
+
+        {isLoading && (
+          <Icon
+            path={spinner}
+            className="absolute right-2 top-2 size-5 animate-spin"
+          />
+        )}
+
+        <ComboboxOptions
+          anchor="bottom start"
+          className={
+            barSize === "small"
+              ? "z-10 max-h-96 w-96 divide-y divide-gray-300 bg-gray-200 [--anchor-max-height:theme(height.96)] empty:hidden dark:divide-gray-900 dark:bg-gray-800"
+              : "z-10 max-h-96 w-full divide-y divide-gray-300 bg-gray-200 [--anchor-max-height:theme(height.96)] empty:hidden dark:divide-gray-900 dark:bg-gray-800"
+          }
+        >
+          {filtered.map((item) => (
+            <div key={item.type}>
+              <h2 className="px-4 py-2 text-sm font-semibold">{item.label}</h2>
+              {item.results.map((result) => (
+                <ComboboxOption
+                  key={result.id}
+                  value={result}
+                  className="cursor-pointer px-4 py-2 data-[focus]:bg-brand-500 data-[focus]:text-inherit dark:data-[focus]:bg-brand-800"
+                  as="div"
+                >
+                  {(item.type === SearchResultType.Provider ||
+                    item.type === SearchResultType.ProviderResource ||
+                    item.type === SearchResultType.ProviderDatasource ||
+                    item.type === SearchResultType.ProviderFunction) && (
+                    <SearchProviderResult result={result} />
+                  )}
+                  {item.type === SearchResultType.Module && (
+                    <SearchModuleResult result={result} />
+                  )}
+                  {item.type === SearchResultType.Other && (
+                    <SearchOtherResult result={result} />
+                  )}
+                </ComboboxOption>
+              ))}
+            </div>
+          ))}
+        </ComboboxOptions>
+      </div>
+    </Combobox>
+  );
+}
+
+Search.defaultProps = {
```

Comment:
`defaultProps` are getting removed from React 19. I'd suggest providing these as default values on the 115th line.